spec/requests/posts/index_spec.rb

```diff
@@ -1,24 +1,58 @@
 require 'rails_helper'
 require 'capybara/rspec'
 
-RSpec.describe 'Post index page', type: :request do
-  it 'displays the posts and their details' do
-    # Create a test user
-    user = User.create(name: 'John Doe', photo: 'valid_photo_url')
+RSpec.describe 'User profile page', type: :feature do
+  before do
+    @user = User.create(name: 'Thomas Heflord',
+                        photo: 'https://media.istockphoto.com/id/1289461328/photo/portrait-of-a-handsome-black-man.jpg?s=612x612&w=0&k=20&c=y_mzB0Tbe5LErNy6pqfY7sz2HiDT7fOAUCwupN3-Bg4=',
+                        posts_counter: 5)
+    @posts = [
+      Post.create(user: @user, title: 'Morality',
+                  text: 'Strive for Personal Growth: Embrace personal growth and strive to become the best',
+                  comments: []),
+      Post.create(user: @user, title: 'Post 2 ',
+                  text: 'Sample text',
+                  comments: []),
+      Post.create(user: @user, title: 'Post 3',
+                  text: 'Practice integrity and honesty in your words and actions. Be true to your principles and',
+                  comments: []),
+      Post.create(user: @user, title: 'Post 4',
+                  text: 'Sample text',
+                  comments: []),
+      Post.create(user: @user, title: 'Post 5',
+                  text: 'Sample text',
+                  comments: [])
+    ]
+  end
+  it 'displays the user profile information and posts' do
+    visit user_path(@user)
+
+    expect(page).to have_content(@user.name)
+    expect(page).to have_selector('img[src="https://media.istockphoto.com/id/1289461328/photo/portrait-of-a-handsome-black-man.jpg?s=612x612&w=0&k=20&c=y_mzB0Tbe5LErNy6pqfY7sz2HiDT7fOAUCwupN3-Bg4="]')

```

Comment:
- [Optional] Dear @TayyabKM, I notice you have written a `src` image value for the image test.  Please be reminded that using hard-written values in a test is not an optimal way of testing our functionalities as the value might change at some point in the future. A better way would be to take get the value from the database and check if it is in the `display/src`. Imagin a user changes their profile picture and your test fails. 

> Something like 
``` ruby 
 expect(page).to have_selector('img[src="#{@user.photo}")
```